Sources/Brave/Frontend/Settings/Tabs/PrivateTabsView.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveStrings
+import Preferences
+import BraveUI
+import Data
+import LocalAuthentication
+
+struct PrivateTabsView: View {
+  enum AuthenticationType {
+    case faceID, touchID, pinCode, noAuthentication
+  }
+  
+  @ObservedObject var privateBrowsingOnly = Preferences.Privacy.privateBrowsingOnly
+  var tabManager: TabManager?
+  
+  private var localAuthenticationType: AuthenticationType {
+    let context = LAContext()
+    
+    if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil) {
+      switch context.biometryType {
+      case .faceID:
+        return .faceID
+      case .touchID:
+        return .touchID
+      default:
+        return .noAuthentication
+      }
+    }
+    
+    var error: NSError?
+    let policyEvaluation = context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error)
+    
+    if policyEvaluation {
+      return .pinCode
+    } else {
+      if (error as? LAError)?.code == .passcodeNotSet {
+        return .noAuthentication
+      }
+    }
+    
+    return .noAuthentication
+  }
+  
+  private var browsingLockTitle: String {
+    var title: String
+    
+    switch localAuthenticationType {
+    case .faceID:
+      title = Strings.TabsSettings.privateBrowsingLockTitleFaceID
+    case .touchID:
+      title = Strings.TabsSettings.privateBrowsingLockTitleTouchID
+    default:
```

Comment:
Is this okay? `default` will return `pinCode` even if `localAuthenticationType` is `noAuthentication`

Reply:
Not just ok, it is also necessary because if it is no authentication that cell is disabled and pincode is shown in menu.